product_filters/models/product.py

```diff
@@ -74,54 +74,56 @@ def _get_async_ranges(self, category):
         ranges_obj = self.env['product.price.ranges'].search([])
         count_dict = {}
         prod_ids = []
-        if category:
-            prod_ids = prod_obj.search(
-                [('public_categ_ids', 'child_of', int(category)),
-                 ('website_published', '=', True)])
-        if prod_ids:
-            for prod in prod_ids:
-                for ran in ranges_obj:
-                    if ran.upper > prod.list_price > ran.lower:
-                        if ran.id in count_dict.keys():
-                            count_dict[ran.id] += 1
-                        else:
-                            count_dict[ran.id] = 1
-                    if ran.id not in count_dict.keys():
-                        count_dict[ran.id] = 0
-            to_jsonfy = [{'id': k, 'qty': count_dict[k]} for k in count_dict]
-            return to_jsonfy
+        if not category:
+            return []
+        prod_ids = prod_obj.search(
+            [('public_categ_ids', 'child_of', int(category)),
+             ('website_published', '=', True)])
+        if not prod_ids:
+            return []
+        for prod in prod_ids:
+            for ran in ranges_obj:
+                if ran.upper > prod.list_price > ran.lower:
+                    if ran.id in count_dict.keys():
```

Comment:
You could avoid use `.keys()`
<img width="800" alt="screen shot 2016-07-08 at 12 35 33 pm" src="https://cloud.githubusercontent.com/assets/6644187/16696032/945dad94-4508-11e6-8e7f-e3cfdc42d12b.png">